# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/manager/ssl/nsICryptoHash.idl
# Commit: 2b647eb982bc
# Full Hash: 2b647eb982bc0068d39fe58c6f59bebe55909343
# Author: Dana Keeler <dkeeler@mozilla.com>
# Date: 2023-06-07 21:43:58
# Regressor Bug: 1834862
# File Overlap Count: 1
# Description:
#   Bug 1834862 - implement nsICryptoHash in rust r=jschanck,supply-chain-reviewers
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D179727
# ==============================================================================

diff -r 7c11166969ff -r 2b647eb982bc security/manager/ssl/nsICryptoHash.idl
--- a/security/manager/ssl/nsICryptoHash.idl	Wed Jun 07 15:45:14 2023 +0000
+++ b/security/manager/ssl/nsICryptoHash.idl	Wed Jun 07 15:56:37 2023 +0000
@@ -3,12 +3,9 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #include "nsISupports.idl"
+
 interface nsIInputStream;
 
-%{C++
-#include "mozilla/AlreadyAddRefed.h"
-%}
-
 /**
  * nsICryptoHash
  * This interface provides crytographic hashing algorithms.
@@ -25,11 +22,11 @@
      * This allows us to use NSS mapping functions like
      * HASH_GetHashOidTagByHashType with these values.
      */
-    const short MD5    = 2;  /* String value: "md5"    */
-    const short SHA1   = 3;  /* String value: "sha1"   */
-    const short SHA256 = 4;  /* String value: "sha256" */
-    const short SHA384 = 5;  /* String value: "sha384" */
-    const short SHA512 = 6;  /* String value: "sha512" */
+    const unsigned long MD5    = 2;  /* String value: "md5"    */
+    const unsigned long SHA1   = 3;  /* String value: "sha1"   */
+    const unsigned long SHA256 = 4;  /* String value: "sha256" */
+    const unsigned long SHA384 = 5;  /* String value: "sha384" */
+    const unsigned long SHA512 = 6;  /* String value: "sha512" */
 
     /**
      * Initialize the hashing object. This method may be
@@ -107,7 +104,6 @@
 };
 
 %{C++
-already_AddRefed<nsICryptoHash> NS_NewCryptoHash();
 nsresult NS_NewCryptoHash(uint32_t aHashType, nsICryptoHash** aOutHasher);
 nsresult NS_NewCryptoHash(const nsACString& aHashType, nsICryptoHash** aOutHasher);
 %}